src/main/java/org/truffleruby/parser/ast/ConstDeclParseNode.java

```diff
@@ -48,6 +48,7 @@
     public ConstDeclParseNode(SourceIndexLength position, String name, INameNode constNode, ParseNode valueNode) {
         super(position, valueNode);
 
+        assert constNode != null || (name != null && !name.isEmpty());
```

Comment:
Should this be a `&&`?
Or it's OK to have null/empty name if `constNode` is non-null?

Reply:
The latter. This is like two parse nodes in one really and you only set one field. They have a comment about that somewhere. I don't want to fix that in this PR.